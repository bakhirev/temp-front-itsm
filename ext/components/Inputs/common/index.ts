export * from './interface';
export * from './hooks';
export * from './utils';
export * from './helpers';
export * from './types';
