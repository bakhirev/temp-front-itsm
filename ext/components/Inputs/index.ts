import { TimePicker } from '../TimePicker';

import { InputText } from './Text';
import { InputPassword } from './Password';
import { InputSuggest } from './Suggest';
import { InputInformer } from './Informer';
import { InputNumber } from './Number';
import { InputMask } from './Mask';
import { InputRange } from './InputRange';
import { InputPhoneCountry } from './PhoneCountry';
export { IInputProps } from './BaseField';
export { IInputMaskProps } from './Mask';
export type { IOptionSuggest } from './Suggest';

export const Input = {
  Text: InputText,
  Password: InputPassword,
  Number: InputNumber,
  Mask: InputMask,
  Suggest: InputSuggest,
  Range: InputRange,
  Informer: InputInformer,
  PhoneCountry: InputPhoneCountry,
  Time: TimePicker,
};
