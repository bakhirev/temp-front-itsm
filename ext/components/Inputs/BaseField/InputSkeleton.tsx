import React, {
  forwardRef,
  Dispatch,
  SetStateAction,
  ChangeEvent,
  FocusEvent,
  KeyboardEvent,
  ReactNode,
  useState,
  FC,
  MutableRefObject,
  RefObject,
  memo,
  useRef,
} from 'react';

import {
  useSuffixHook,
  formatValue,
  formatValueWithSeparator,
  joinIntegerWithFraction,
} from '../common';
import type { Status, Size, Type, Mask } from '../common/types';
import { Body1Short, Body2Short } from '../../Typography';
import { KEY_CODES } from '../../common/key-codes';
import { DEFAULT_WIDTH } from '../constants';

import EditableCoinsWrapper from './editable-coins';
import {
  InputWrapper,
  InputComponent,
  IconWrapper,
  UserIcon,
  Suffix,
  SeparatorDot,
  CountryPhoneInputBox,
  CountrySelectWrapper,
} from './StyledComponents';

export interface ISkeletonProps {
  onChange?: (event: ChangeEvent<HTMLInputElement>, value: string) => void;
  focused: boolean;
  setFocused: Dispatch<SetStateAction<boolean>>;
  value?: any;
  placeholder?: string;
  id?: string;
  icon?: ReactNode;
  type?: Type;
  inputName?: string;
  onClick?: (event: any) => void | undefined;
  iconServices?: ReactNode;
  size?: Size;
  status?: Status;
  width?: string | number;
  disabled?: boolean;
  onBlur?: (event: FocusEvent<HTMLInputElement>, value: string) => void;
  onFocus?: (event: FocusEvent<HTMLInputElement>) => void;
  onKeyDown?: (event: KeyboardEvent<HTMLInputElement>) => void;
  activeInformer?: boolean;
  maxLength?: number;
  mask?: Mask | object | string | any;
  suffix?: string;
  iconIsActive?: boolean;
  setIconIsActive?: Dispatch<SetStateAction<boolean>>;
  number?: boolean;
  range?: boolean;
  withCoins?: boolean;
  phoneCountry?: boolean;
  countryIcon?: any;
  selectIcon?: any;
  onCountrySelectOpen?: Function;
  countrySelectComponent?: any;
  onCountrySelect?: Function;
  ref: MutableRefObject<HTMLInputElement> | null | RefObject<HTMLInputElement> | undefined;
  defaultValue?: any;
}

export const InputSkeleton: FC<ISkeletonProps> = memo(
  forwardRef(
    (
      {
        status = 'default',
        width = DEFAULT_WIDTH,
        disabled,
        focused,
        setFocused,
        size = 'big',
        onChange,
        icon,
        type,
        iconServices,
        onFocus,
        value: rawValue,
        placeholder,
        onBlur,
        onKeyDown,
        onClick,
        inputName,
        activeInformer,
        maxLength,
        id,
        suffix,
        iconIsActive,
        mask,
        setIconIsActive,
        number,
        range,
        withCoins,
        phoneCountry,
        countryIcon: CountryIcon,
        selectIcon: SelectIcon,
        onCountrySelectOpen,
        countrySelectComponent: CountrySelectComponent,
        onCountrySelect,
        defaultValue,
      },
      ref: any
    ) => {
      const hasFraction = number && rawValue && rawValue.includes('.');
      const [integer, fraction] = hasFraction ? rawValue.split('.') : [rawValue, '00'];

      const [integerWidth, refNode, overflowed] = useSuffixHook(
        formatValueWithSeparator(integer),
        Boolean(suffix) || withCoins,
        size,
        ref
      );
      const [coins, setCoins] = useState(fraction);
      const coinsRef: MutableRefObject<HTMLDivElement | null> = useRef(null);

      const focusCoinsInput = () => {
        coinsRef?.current && coinsRef.current.focus();
      };

      const SuffixLabel = size === 'micro' ? Body2Short : Body1Short;

      const handleMemomizeChange = (event: ChangeEvent<HTMLInputElement>): void => {
        const { value } = event.target;

        !range && setFocused(true);
        !number
          ? onChange && onChange(event, value)
          : onChange &&
            onChange(
              event,
              withCoins ? joinIntegerWithFraction(formatValue(value), coins) : formatValue(value)
            );
      };

      const handleFocus = (event: FocusEvent<HTMLInputElement>): void => {
        setFocused(true);
        onFocus && onFocus(event);
      };

      const setIntegerInputFocused = (event: any) => {
        event.preventDefault() && event.stopPropagation();
        event.currentTarget.parentNode.children[0].focus();
      };

      const handleBlur = (event: FocusEvent<HTMLInputElement>): void => {
        const { value } = event.target;
        setFocused(false);

        onBlur && onBlur(event, value);
      };

      const handleKeyDown = (event: any): void => {
        const selection = event.target.selectionStart;
        if (event.keyCode === KEY_CODES.RIGHT && selection === event.target.value.length) {
          focusCoinsInput();
          return;
        }

        onKeyDown && onKeyDown(event);
        const { value } = event.target;

        if (!disabled && event.keyCode === KEY_CODES.ENTER) {
          onChange && onChange(event, value);
          event.target.blur();
        }
      };

      const handleMouseLeave = () => {
        iconIsActive && setIconIsActive && setIconIsActive(false);
      };

      const onCoinsChange = (event: any) => {
        setCoins(event.target.value);
        const newRawValue = integer ? `${integer}.${event.target.value || '00'}` : '';
        onChange && onChange(event, newRawValue);
      };

      const countrySelectHandler = (option) => {
        onCountrySelect && onCountrySelect(option);
        if (ref.current) {
          ref.current.focus();
          setFocused(true);
        }
      };

      const [selectMenuOpen, setSelectMenuOpen] = useState<boolean>(false);

      const shouldRenderCoins = number && withCoins && rawValue && !defaultValue;
      const shouldRenderBasicInput = (!mask || number) && !CountrySelectComponent && onChange;
      const shouldRenderUserIcon = icon && status !== 'error';
      const shouldRenderSuffix = suffix && Boolean(rawValue.length) && !overflowed;
      const shouldRenderIconWrapper = icon || iconServices;
      const shouldRenderUncontrollInput = defaultValue && !onChange;

      return (
        <InputWrapper
          onMouseLeave={handleMouseLeave}
          focused={focused}
          disabled={disabled}
          status={status}
          size={size}
          range={range}
        >
          {shouldRenderUncontrollInput && (
            <InputComponent
              id={id}
              // eslint-disable-next-line jsx-a11y/no-autofocus
              autoFocus={focused}
              name={inputName}
              ref={number ? refNode : ref}
              $size={size}
              disabled={disabled}
              defaultValue={defaultValue}
              placeholder={placeholder}
              onFocus={handleFocus}
              onBlur={handleBlur}
              focused={focused}
              type={type}
              autoComplete="off"
              maxLength={maxLength}
            />
          )}
          {CountrySelectComponent && (
            <>
              <CountryPhoneInputBox>
                <CountrySelectWrapper
                  size={size}
                  onClick={(e) => {
                    e.preventDefault();
                    const open = selectMenuOpen;
                    setSelectMenuOpen(!open);
                    setFocused(!open);
                  }}
                >
                  <CountrySelectComponent
                    onCountrySelect={countrySelectHandler}
                    disabled={disabled}
                    focused={focused}
                    size={size}
                    onMenuOpen={() => setFocused(true)}
                    onMenuClose={() => setFocused(false)}
                    menuWidth={width}
                    menuIsOpen={selectMenuOpen}
                  />
                </CountrySelectWrapper>
                <InputComponent
                  value={rawValue}
                  $size={size}
                  focused={focused}
                  disabled={disabled}
                  id={id}
                  // eslint-disable-next-line jsx-a11y/no-autofocus
                  autoFocus={focused}
                  name={inputName}
                  ref={ref}
                  onChange={handleMemomizeChange}
                  onFocus={handleFocus}
                  onBlur={handleBlur}
                  onKeyDown={handleKeyDown}
                  type={type}
                  autoComplete="off"
                  onClick={onClick}
                  maxLength={maxLength}
                  withSelectComponent
                />
              </CountryPhoneInputBox>
            </>
          )}
          {shouldRenderBasicInput && (
            <InputComponent
              id={id}
              // eslint-disable-next-line jsx-a11y/no-autofocus
              autoFocus={focused}
              name={inputName}
              ref={number ? refNode : ref}
              $size={size}
              disabled={disabled}
              placeholder={placeholder}
              value={number ? formatValueWithSeparator(integer) : rawValue}
              onChange={handleMemomizeChange}
              onFocus={handleFocus}
              onBlur={handleBlur}
              onKeyDown={handleKeyDown}
              focused={focused}
              type={type}
              autoComplete="off"
              onClick={onClick}
              maxLength={maxLength}
            />
          )}
          {shouldRenderCoins && (
            <SeparatorDot position={integerWidth} disabled={disabled} size={size}>
              {'.'}
            </SeparatorDot>
          )}
          {shouldRenderCoins && (
            <EditableCoinsWrapper
              ref={coinsRef}
              html={coins}
              setFocusedStyle={setFocused}
              setIntegerFocused={(event) => {
                setFocused(true);
                setIntegerInputFocused(event);
              }}
              position={integerWidth}
              onChange={onCoinsChange}
              disabled={disabled}
              dimension={size}
            />
          )}
          {shouldRenderSuffix && (
            <Suffix
              onMouseDown={setIntegerInputFocused}
              size={size}
              disabled={disabled}
              position={integerWidth}
              withCoins={shouldRenderCoins}
            >
              <SuffixLabel>{suffix}</SuffixLabel>
            </Suffix>
          )}
          {shouldRenderIconWrapper && (
            <IconWrapper onMouseDown={(e) => e.preventDefault()}>
              {shouldRenderUserIcon && <UserIcon disabled={disabled}>{icon}</UserIcon>}
              {iconServices}
            </IconWrapper>
          )}
        </InputWrapper>
      );
    }
  )
);

InputSkeleton.displayName = 'InputSkeleton';
