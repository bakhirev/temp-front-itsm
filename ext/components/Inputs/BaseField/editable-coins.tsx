import React, { KeyboardEvent } from 'react';

import { KEY_CODES } from '../../common';
import type { Size } from '../common/types';

import { Coins } from './StyledComponents';

export interface IEditableCoinsProps {
  onChange: (event: any) => void;
  html: string;
  position: number;
  disabled?: boolean;
  setIntegerFocused: Function;
  setFocusedStyle: Function;
  dimension: Size;
}

class EditableCoins extends React.Component<IEditableCoinsProps & { innerRef: any }> {
  elementRef: any;
  lastHtml = '';

  constructor(props) {
    super(props);
    this.elementRef = props.innerRef;
  }

  shouldComponentUpdate = (nextProps) => {
    return (
      nextProps.html !== this.elementRef?.current?.innerHTML ||
      this.props.position !== nextProps.position ||
      this.props.dimension !== nextProps.dimension ||
      this.props.disabled !== nextProps.disabled
    );
  };

  emitChange = () => {
    const html = this.elementRef?.current?.innerHTML;
    const { onChange } = this.props;

    if (onChange && html !== this.lastHtml) {
      onChange({ target: { value: html } });
    }
    this.lastHtml = html;
  };

  onBlur = (event) => {
    const { innerText } = event.target;
    const { onChange } = this.props;

    if (onChange && innerText.length < 2) {
      onChange({ target: { value: '00' } });
      this.lastHtml = '00';
    }

    this.props.setFocusedStyle(false);
  };

  onKeyDown = (event: KeyboardEvent<HTMLElement>): void | boolean => {
    const { disabled, setIntegerFocused } = this.props;

    if (disabled) {
      event.preventDefault();
      return false;
    }

    const { keyCode, shiftKey } = event;
    const { innerText } = event.target as HTMLElement;
    const selection = window.getSelection();
    const focusInTheBeginning = selection?.focusOffset === 0;

    if (focusInTheBeginning && keyCode === KEY_CODES.LEFT) {
      this.onBlur(event);
      setIntegerFocused && setIntegerFocused(event);
      return true;
    }

    if (
      // non input key codes
      [
        KEY_CODES.LEFT,
        KEY_CODES.RIGHT,
        KEY_CODES.BACKSPACE,
        KEY_CODES.DELETE,
        KEY_CODES.SHIFT,
        KEY_CODES.ALT,
        KEY_CODES.CTRL,
        KEY_CODES.TAB,
        KEY_CODES.HOME,
        KEY_CODES.END,
      ].includes(keyCode)
    ) {
      return true;
    }

    if (
      innerText.length === 2 ||
      !(KEY_CODES.ZERO <= keyCode && keyCode <= KEY_CODES.NINE) ||
      shiftKey
    ) {
      event.preventDefault();
      return false;
    }
  };

  render() {
    return (
      <Coins
        ref={this.props.innerRef}
        key={Date()}
        position={this.props.position}
        onInput={this.emitChange}
        onBlur={this.onBlur}
        onFocus={() => this.props.setFocusedStyle(true)}
        onKeyDown={this.onKeyDown}
        dimension={this.props.dimension}
        contentEditable
        dangerouslySetInnerHTML={{ __html: this.props.html }}
        disabled={this.props.disabled}
      />
    );
  }
}

export default React.forwardRef((props: IEditableCoinsProps, ref: any) => (
  <EditableCoins innerRef={ref} {...props} />
));
