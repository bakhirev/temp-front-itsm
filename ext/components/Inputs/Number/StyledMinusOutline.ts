import styled, { css } from 'styled-components';

import { ReactComponent as MinusOutline } from '../../Icons/service/MinusOutline.svg';
import { getDefaultIconStyled, IStyledIconDefaultProps } from '../common';
import { DEFAULT_THEME } from '../../common';

export const StyledMinusOutline = styled(MinusOutline)<IStyledIconDefaultProps>`
  ${({ theme, disabled }) => css`
    ${getDefaultIconStyled(theme, disabled)}
  `}
`;

StyledMinusOutline.defaultProps = {
  theme: DEFAULT_THEME,
};
