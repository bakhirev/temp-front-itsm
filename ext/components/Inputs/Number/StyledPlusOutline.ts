import styled, { css } from 'styled-components';

import { ReactComponent as PlusOutline } from '../../Icons/service/PlusOutline.svg';
import { DEFAULT_THEME } from '../../common';
import { getDefaultIconStyled, IStyledIconDefaultProps } from '../common';

export const StyledPlusOutline = styled(PlusOutline)<IStyledIconDefaultProps>`
  ${({ theme, disabled }) => css`
    ${getDefaultIconStyled(theme, disabled)}
  `}
`;

StyledPlusOutline.defaultProps = {
  theme: DEFAULT_THEME,
};
