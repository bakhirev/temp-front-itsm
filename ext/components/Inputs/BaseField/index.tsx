import React, { useState, Dispatch, SetStateAction, ReactNode, FC } from 'react';

import { createTooltipHOC } from '../../TooltipHOC';
import { Body2Short } from '../../Typography';
import { IInputDefaultProps, useTooltipHook } from '../common';
import type { Type } from '../common/types';
import { DEFAULT_WIDTH } from '../constants';

import { InputSkeleton } from './InputSkeleton';
import { Label, Wrapper, Separator, StyledAdditionalText } from './StyledComponents';

export interface IInputProps extends IInputDefaultProps {
  /** Коллбэк на клик */
  onClick?: (event: any) => void | undefined;
  /** иконки из Figma , которые прокидываются в инпут */
  iconServices?: ReactNode;
  /** Отображение подсказки при переполнении инпута. По умолчанию false */
  withTooltip?: boolean;
  /** Тип инпута */
  type?: Type;
  /** Активна ли иконка в инпуте */
  iconIsActive?: boolean;
  /** Функция для переключения active icon */
  setIconIsActive?: Dispatch<SetStateAction<boolean>>;
  /** Использование числового инпута */
  number?: boolean;
  /** Использование Input Range */
  range?: boolean;
  /** Активен ли информер в инпуте */
  activeInformer?: boolean;
  /** Значение инпута */
  value?: any;
  /** Флаг ввода дробной части */
  withCoins?: boolean;
  phoneCountry?: boolean;
  countryIcon?: any;
  selectIcon?: any;
  onCountrySelectOpen?: Function;
  countrySelectComponent?: any;
  onCountrySelect?: Function;
  /** Дефолтное value */
  defaultValue?: any;
}

export const Input: FC<IInputProps> = ({
  disabled,
  label,
  additionalText,
  status = 'default',
  className,
  dataTestId,
  withTooltip = false,
  value = '',
  width = DEFAULT_WIDTH,
  refInput,
  tooltipContainer,
  ...props
}) => {
  const [focused, setFocused] = useState(false);
  const { tooltip, ref } = useTooltipHook(refInput, withTooltip, value);

  const Input = tooltip && !focused ? createTooltipHOC(InputSkeleton) : InputSkeleton;
  const inputElement = (
    <Input
      tooltip={value}
      container={tooltipContainer}
      display="block"
      disabled={disabled}
      focused={focused}
      setFocused={setFocused}
      status={status}
      value={value}
      ref={ref}
      width={width}
      {...props}
    />
  );

  return (
    <Wrapper
      disabled={disabled}
      focused={focused}
      className={className}
      data-test-id={dataTestId}
      width={width}
    >
      <Label>
        {label && (
          <>
            <Body2Short>{label}</Body2Short>
            <Separator />
          </>
        )}
        {inputElement}
      </Label>
      {additionalText && (
        <StyledAdditionalText disabled={disabled} status={status} focused={focused}>
          <Body2Short>{additionalText}</Body2Short>
        </StyledAdditionalText>
      )}
    </Wrapper>
  );
};
