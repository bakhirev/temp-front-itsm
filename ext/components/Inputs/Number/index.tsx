import React, { useState, useCallback, FC, useEffect } from 'react';

import { Input } from '../BaseField';
import { INCREMENT, DECREMENT, DEFAULT_MAX_INPUT_LENGTH } from '../constants';
import { formatValue, getIconSize } from '../common';
import type { IInputDefaultProps } from '../common/interface';

import { getIntegerMaxValue, getResultValue } from './utils';
import { StyledPlusOutline } from './StyledPlusOutline';
import { StyledMinusOutline } from './StyledMinusOutline';
import { IconWrapper } from './IconWrapper';

export interface IInputNumberProps extends IInputDefaultProps {
  /** Значение инпута */
  value?: string;
  /** Шаг прибавления или убавления значения */
  step?: number;
  /** Суффикс */
  suffix?: string;
  /** Максимальное количество символов в поле ввода */
  maxLength?: number;
  /** Флаг отключения иконок прибавления или убавления */
  withoutIconServices?: boolean;
  /** Флаг ввода дробной части */
  withCoins?: boolean;
}

export const InputNumber: FC<IInputNumberProps> = ({
  value = '',
  disabled,
  size = 'big',
  onChange,
  status,
  step = 100,
  maxLength = DEFAULT_MAX_INPUT_LENGTH,
  withoutIconServices,
  withCoins,
  ...props
}) => {
  const [iconIsActive, setIconIsActive] = useState<boolean>(false);
  const [plusDisabled, setPlusDisabled] = useState<boolean>(!!disabled);
  const [minusDisabled, setMinusDisabled] = useState<boolean>(!!disabled);
  const maxValue = getIntegerMaxValue(maxLength);
  const minValue = 0;

  useEffect(() => {
    const cantDecrement = minValue > +formatValue(value) - step;
    const cantIncrement = maxValue < +formatValue(value) + step;
    cantDecrement !== minusDisabled && setMinusDisabled(cantDecrement);
    cantIncrement !== plusDisabled && setPlusDisabled(cantIncrement);
  }, [value]);

  const handleDecrement = useCallback(
    (event: any) => {
      onChange?.(event, getResultValue(value, step, DECREMENT));

      setIconIsActive(true);
    },
    [onChange, step, value]
  );

  const handleIncrement = useCallback(
    (event: any) => {
      onChange?.(event, getResultValue(value, step, INCREMENT));

      setIconIsActive(true);
    },
    [onChange, step, value]
  );

  const iconSize = getIconSize(size);
  const iconServices = (
    <IconWrapper size={size}>
      <StyledMinusOutline
        onClick={!minusDisabled && !disabled ? handleDecrement : undefined}
        width={iconSize}
        height={iconSize}
        disabled={disabled || minusDisabled}
      />
      <StyledPlusOutline
        onClick={!plusDisabled && !disabled ? handleIncrement : undefined}
        width={iconSize}
        height={iconSize}
        disabled={disabled || plusDisabled}
      />
    </IconWrapper>
  );

  return (
    <Input
      disabled={disabled}
      size={size}
      status={status}
      onChange={onChange}
      value={value}
      iconServices={withoutIconServices ? null : iconServices}
      iconIsActive={iconIsActive}
      setIconIsActive={setIconIsActive}
      maxLength={maxLength}
      number
      withCoins={withCoins}
      {...props}
    />
  );
};
