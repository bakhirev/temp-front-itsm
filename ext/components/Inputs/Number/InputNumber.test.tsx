import React from 'react';
import { shallow } from 'enzyme';

import { runCommonInputTest, runCommonInputOnChangeTest } from '../common/test';
import { Input } from '../../Inputs';
import { InputSkeleton } from '../BaseField/InputSkeleton';

import { StyledMinusOutline } from './StyledMinusOutline';
import { StyledPlusOutline } from './StyledPlusOutline';

jest.mock('../../common', () => ({
  DEFAULT_THEME: {},
  Z_INDEXES: {
    INFORMER: 777,
  },
}));

describe('Input.Number', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  const spyOn = jest.fn();

  const step = 1000;

  runCommonInputTest(Input.Number);
  runCommonInputOnChangeTest(Input.Number, '1000', '1000');
  runCommonInputOnChangeTest(Input.Number, 'qwerty', '');
  runCommonInputOnChangeTest(Input.Number, '1 000', '1000');

  it('should call onChange increment value when user clicks on plus icon', () => {
    const wrapper = shallow(<Input.Number size={'big'} value="0" step={step} onChange={spyOn} />);

    const icon = wrapper.dive().find(InputSkeleton).dive().find(StyledPlusOutline);
    icon.simulate('click');
    expect(spyOn).toBeCalledTimes(1);
    expect(spyOn).toBeCalledWith(undefined, '1000');
  });

  it('should call onChange decrement value when user clicks on minus icon', () => {
    const wrapper = shallow(
      <Input.Number size={'big'} value="3000" step={step} onChange={spyOn} />
    );

    const icon = wrapper.dive().find(InputSkeleton).dive().find(StyledMinusOutline);
    icon.simulate('click');
    expect(spyOn).toBeCalledTimes(1);
    expect(spyOn).toBeCalledWith(undefined, '2000');
  });

  it('should call onChange decrement value when user clicks on minus icon and value === 0', () => {
    const wrapper = shallow(<Input.Number size={'big'} onChange={spyOn} value="0" step={step} />);

    const icon = wrapper.dive().find(InputSkeleton).dive().find(StyledMinusOutline);
    icon.simulate('click');
    expect(spyOn).toBeCalledTimes(1);
    expect(spyOn).toBeCalledWith(undefined, '0');
  });
});
