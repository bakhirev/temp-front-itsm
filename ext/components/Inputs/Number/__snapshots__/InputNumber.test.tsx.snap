// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.Number should render component 1`] = `
<Component
  iconIsActive={false}
  iconServices={
    <ForwardRef(styled.div)
      size="big"
    >
      <ForwardRef(styled.div)
        disabled={false}
        height={24}
        onClick={[Function]}
        theme={Object {}}
        width={24}
      />
      <ForwardRef(styled.div)
        disabled={false}
        height={24}
        onClick={[Function]}
        theme={Object {}}
        width={24}
      />
    </ForwardRef(styled.div)>
  }
  maxLength={16}
  number={true}
  onChange={[Function]}
  setIconIsActive={[Function]}
  size="big"
  value=""
/>
`;

exports[`Input.Number should render component with clearable field, icon and name 1`] = `
<Component
  clearable={true}
  icon={<div />}
  iconIsActive={false}
  iconServices={
    <ForwardRef(styled.div)
      size="micro"
    >
      <ForwardRef(styled.div)
        disabled={false}
        height={20}
        onClick={[Function]}
        theme={Object {}}
        width={20}
      />
      <ForwardRef(styled.div)
        disabled={false}
        height={20}
        onClick={[Function]}
        theme={Object {}}
        width={20}
      />
    </ForwardRef(styled.div)>
  }
  inputName="test"
  label="Введите ваше имя"
  maxLength={16}
  number={true}
  onChange={[Function]}
  setIconIsActive={[Function]}
  size="micro"
  value=""
/>
`;

exports[`Input.Number should render component with error status 1`] = `
<Component
  additionalText="Ошибка"
  iconIsActive={false}
  iconServices={
    <ForwardRef(styled.div)
      size="big"
    >
      <ForwardRef(styled.div)
        disabled={false}
        height={24}
        onClick={[Function]}
        theme={Object {}}
        width={24}
      />
      <ForwardRef(styled.div)
        disabled={false}
        height={24}
        onClick={[Function]}
        theme={Object {}}
        width={24}
      />
    </ForwardRef(styled.div)>
  }
  label=""
  maxLength={16}
  number={true}
  onChange={[Function]}
  setIconIsActive={[Function]}
  size="big"
  status="error"
  value=""
/>
`;

exports[`Input.Number should render component with success status 1`] = `
<Component
  additionalText="Успех"
  iconIsActive={false}
  iconServices={
    <ForwardRef(styled.div)
      size="big"
    >
      <ForwardRef(styled.div)
        disabled={false}
        height={24}
        onClick={[Function]}
        theme={Object {}}
        width={24}
      />
      <ForwardRef(styled.div)
        disabled={false}
        height={24}
        onClick={[Function]}
        theme={Object {}}
        width={24}
      />
    </ForwardRef(styled.div)>
  }
  label="Введите ваше имя"
  maxLength={16}
  number={true}
  onChange={[Function]}
  setIconIsActive={[Function]}
  size="big"
  status="success"
  value=""
/>
`;
